src/utils/router.ts

```diff
@@ -0,0 +1,16 @@
+import { matchPath } from 'react-router-dom'
+import { BASE_SAFE_ROUTE } from 'src/routes/routes'
+import { checksumAddress } from './checksumAddress'
+import { history } from '../routes/routes'
+
+export const safeAddressFromUrl = (): string => {
+  const match = matchPath<{ safeAddress: string }>(history.location.pathname, {
+    path: BASE_SAFE_ROUTE,
+  })
+
+  if (match) {
+    return checksumAddress(match.params.safeAddress).toString()
```

Comment:
This'll throw if the address is invalid. I think it's fine to show a crash page in this case but we wouldn't want this logged on Sentry, probably?